src/main/java/org/opensearch/security/dlic/rest/api/CertificatesApiAction.java

```diff
@@ -0,0 +1,119 @@
+
+    private final static Logger LOGGER = LogManager.getLogger(CertificatesApiAction.class);
+
+    private static final List<Route> ROUTES = addRoutesPrefix(
+        ImmutableList.of(new Route(RestRequest.Method.GET, "/certificates"), new Route(RestRequest.Method.GET, "/certificates/{nodeId}")),
+        PLUGIN_API_ROUTE_PREFIX
+    );
+
+    public CertificatesApiAction(
+        final ClusterService clusterService,
+        final ThreadPool threadPool,
+        final SecurityApiDependencies securityApiDependencies
+    ) {
+        super(Endpoint.SSL, clusterService, threadPool, securityApiDependencies);
+        this.requestHandlersBuilder.configureRequestHandlers(this::securitySSLCertsRequestHandlers);
+    }
+
+    @Override
+    public List<Route> routes() {
+        return ROUTES;
+    }
+
+    @Override
+    public String getName() {
+        return "HTTP and Transport Certificates Actions";
+    }
+
+    @Override
+    protected CType getConfigType() {
+        return null;
+    }
+
+    @Override
+    protected void consumeParameters(RestRequest request) {
+        request.param("nodeId");
+        request.param("cert_type");
+    }
+
+    private void securitySSLCertsRequestHandlers(RequestHandler.RequestHandlersBuilder requestHandlersBuilder) {
+        requestHandlersBuilder.withAccessHandler(this::accessHandler)
+            .allMethodsNotImplemented()
+            .verifyAccessForAllMethods()
+            .override(
+                RestRequest.Method.GET,
+                (channel, request, client) -> client.execute(
+                    CertificatesActionType.INSTANCE,
+                    new CertificatesInfoNodesRequest(
+                        request.param("cert_type", CertificatesInfoNodesRequest.ALL_CERT_TYPE),
+                        true,
+                        request.paramAsStringArrayOrEmptyIfAll("nodeId")
+                    ),
+                    new ActionListener<>() {
+                        @Override
+                        public void onResponse(final CertificatesNodesResponse response) {
+                            ok(channel, (builder, params) -> {
+                                builder.startObject();
+                                RestActions.buildNodesHeader(builder, channel.request(), response);
+                                builder.field("cluster_name", response.getClusterName().value());
+                                response.toXContent(builder, channel.request());
+                                builder.endObject();
+                                return builder;
+                            });
+                        }
+
+                        @Override
+                        public void onFailure(Exception e) {
+                            LOGGER.error("Cannot load SSL certificates info due to", e);
+                            internalSeverError(channel, "Cannot load SSL certificates info " + e.getMessage() + ".");
```

Comment:
nit: spelling `internalServerError`

Reply:
Oops. I will prepare a follow PR since it uses in multiple places :-) you you don't mind.

Reply:
Fixed here: https://github.com/opensearch-project/security/pull/4305